Java/LFU.java

```diff
@@ -0,0 +1,144 @@
+package lfu;
+import java.util.LinkedList;
+import java.util.ListIterator;
+import java.util.HashMap;
+
+/**
+ * @author salman
```

Comment:
Since this is an instanciable class, you should use a singular noun describing what one instance of this class represents. For example 

An element in cache that is least frequently used.

Honestly, I am skeptical about the name and responsibilities of this class.